crates/sol-types/tests/doctests/function_like.rs

```diff
@@ -25,9 +25,31 @@ sol! {
     /// doc comment, but this might change in the future.
     ///
     /// The [`SolCall`] implementation may be used to decode the return values
-    /// via [`SolCall::decode_returns`]. The return value is a tuple of the
-    /// return types, e.g. `returns (uint256)`  will have a return type of
-    /// `(sol_data::Uint<256>,)` and a decoded type of `(U256,)`.
+    /// via [`SolCall::decode_returns`]. The return value is a struct containing
+    /// the return values, in the same order as the function definition. Unnamed
+    /// return values will be named based on their position, e.g. `_0`, `_1`,
+    /// like the arguments.
+    ///
+    /// For example the following input:
+    ///  `function foo(uint256 a, uint256 b) external view returns (uint256);`
```

Comment:
this is the doc comment's function

Reply:
yeah the rest of it was generic about macro functionality so i decided to be as explicit as possible